src/ReplaySubject.ts

```diff
@@ -90,12 +116,13 @@ export class ReplaySubject<T> extends Subject<T> {
       spliceCount = Math.max(spliceCount, eventsCount - _bufferSize);
     }
 
-    if (spliceCount > 0) {
-      _events.splice(0, spliceCount);
+    for (let i = 0; i < spliceCount; i++) {
+      _events.shift();
     }
```

Comment:
After reviewing perf, i think we should return it to `splice` because of it's consistent performance, even if `shift` is faster on smaller arrays.